controller/internal/enforcer/envoyproxy/authz/server.go

```diff
@@ -0,0 +1,501 @@
+				if httpReqHeaders != nil {
+					aporetoKey, _ = httpReqHeaders[aporetoKeyHeader]
+					aporetoAuth, _ = httpReqHeaders[aporetoAuthHeader]
+				}
+			}
+		}
+	}
+
+	flow := newIngressFlowRecord(s.puID, pctx, source, dest, httpReq)
+	defer func() {
+		zap.L().Debug("ext_authz ingress: collecting flow event", zap.String("puID", s.puID), zap.Any("flow", flow))
+		s.collector.CollectFlowEvent(flow)
+	}()
+
+	// if the request is lacking our stuff, we send a 400 back
+	if aporetoAuth == "" || aporetoKey == "" {
+		zap.L().Warn("ext_authz ingress: request missing Aporeto HTTP headers: x-aporeto-key and/or x-aporeto-auth", zap.String("puID", s.puID), zap.String("sourceAddress", sourceAdress))
+		flow.DropReason = "missing Aporeto HTTP headers"
+		return createDeniedCheckResponse(rpc.INVALID_ARGUMENT, envoy_type.StatusCode_BadRequest, "missing headers X-APORETO-KEY or X-APORETO-AUTH"), nil
+	}
+
+	// now we can see if we can decode our claims
+	srcid, scopes, profile, err := s.verifier.ParseToken(aporetoAuth, aporetoKey)
+	if err != nil {
+		zap.L().Debug("ext_authz ingress: failed to parse Aporeto token", zap.String("puID", s.puID), zap.Error(err))
+		flow.DropReason = "failed to parse Aporeto token"
+		return createDeniedCheckResponse(rpc.PERMISSION_DENIED, envoy_type.StatusCode_Forbidden, "failed to parse Aporeto token"), nil
+	}
+	zap.L().Debug("ext_authz ingress: parsed Aporeto token", zap.String("puID", s.puID), zap.String("srcid", srcid), zap.Strings("scopes", scopes), zap.Strings("profile", profile))
+	flow.Source.ID = srcid
+
+	observedPolicy, netPolicy := pctx.SearchRcvRules(policy.NewTagStoreFromSlice(profile))
+	flow.Action = netPolicy.Action
+	flow.PolicyID = netPolicy.PolicyID
+	flow.ObservedAction = observedPolicy.Action
+	flow.ObservedPolicyID = observedPolicy.PolicyID
+
+	if netPolicy.Action.Rejected() {
+		zap.L().Debug("ext_authz ingress: Access *NOT* authorized by network policy", zap.String("puID", s.puID))
+		flow.DropReason = "access not authorized by network policy"
+		return createDeniedCheckResponse(rpc.PERMISSION_DENIED, envoy_type.StatusCode_Forbidden, "Access not authorized by network policy"), nil
+	}
+
+	// otherwise we are just going to allow it for now
+	zap.L().Debug("ext_authz ingress: Access authorized by network policy", zap.String("puID", s.puID))
+	return &ext_authz_v2.CheckResponse{
+		Status: &rpc.Status{
+			Code: int32(rpc.OK),
+		},
+		HttpResponse: &ext_authz_v2.CheckResponse_OkResponse{
+			OkResponse: &ext_authz_v2.OkHttpResponse{},
+		},
+	}, nil
+}
+
+// egressCheck implements the AuthorizationServer for egress connections
+func (s *Server) egressCheck(ctx context.Context, checkRequest *ext_authz_v2.CheckRequest) (*ext_authz_v2.CheckResponse, error) {
+	zap.L().Debug("ext_authz egress: checkRequest", zap.String("puID", s.puID), zap.String("checkRequest", checkRequest.String()))
+
+	// get the PU context
+	pctxRaw, err := s.puContexts.Get(s.puID)
+	if err != nil {
```

Comment:
same here. We need one more local to discover the destination service context, from there the API specification, and from there the decision is made on whether this is external or internal, public or not, and so on.

Reply:
the destination IP and port contains the IP and port as used outgoing by Envoy. That means: pod IP + the port as specified in the pod template (yes, Istio _requires_ you to fill out that part of your yamls)